### PR TITLE
EW-694 Common Cartridge Course Import

### PR DESCRIPTION
**Pull request by psachmann**
# Short Description
Adding a new modal and import/upload dialog to allow users the import of courses from Common Cartridge files.

## Links to Ticket and related Pull-Requests
- <https://ticketsystem.dbildungscloud.de/browse/EW-694>
- <https://github.com/hpi-schul-cloud/schulcloud-server/pull/4700>
- <https://github.com/hpi-schul-cloud/dof_app_deploy/pull/763>

## Deployment
Environment variables `FEATURE_COMMON_CARTRIDGE_IMPORT_ENABLED` must be set to `true`.

## Screenshots of UI changes
Upload Button
Unselected Upload File
Selected Upload File

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by Fshmit — APPROVED** [review]

**Review by mkreuzkam-cap — DISMISSED** [review]

File: src/store/rooms.unit.ts [hunk 1]
```diff
@@ -333,6 +334,56 @@ describe("rooms module", () => {
...
+
+			it("should call set loading", async () => {
+				const course = new File([], "my-course.zip");
+				const roomsModule = new RoomsModule({});
+				const setAlertDataMock = jest.spyOn(roomsModule, "setAlertData");
+
+				await roomsModule.uploadCourse(course);
+
+				expect(setAlertDataMock).toHaveBeenCalledTimes(1);
+				expect(setAlertDataMock).toHaveBeenCalledWith({
```
Comment (mkreuzkam-cap) [inline src/store/rooms.unit.ts:359]: This fails. I think the mocking of the API is missing and the text should be "pages.rooms.ccImportCourse.success"

Comment (psachmann) [reply src/store/rooms.unit.ts:365]: Fixed the test.

**Review by mkreuzkam-cap — DISMISSED** [review]

**Review by SevenWaysDP — DISMISSED** [review]

File: src/store/rooms.unit.ts [hunk 2]
```diff
@@ -315,6 +316,62 @@ describe("rooms module", () => {
 				expect(setBusinessErrorMock).toHaveBeenCalled();
 			});
 		});
+
+		describe("uploadCourse", () => {
```
Comment (mkreuzkam-cap) [inline src/store/rooms.unit.ts:320]: This method no longer exists in RoomModule. Remove these tests

**Review by mkreuzkam-cap — CHANGES_REQUESTED** [review]

File: src/store/rooms.unit.ts (hunk 2 above)
Comment (psachmann) [reply src/store/rooms.unit.ts:320]: Done.

File: src/components/templates/RoomWrapper.vue [hunk 3]
```diff
@@ -66,40 +78,68 @@ export default Vue.extend({
...
+							},
+						});
+					}
+
+					if (
+						envConfigModule.getEnv
+							.FEATURE_COMMON_CARTRIDGE_COURSE_IMPORT_ENABLED
+					) {
+						fabItems.actions.push({
+							label: this.$t("pages.rooms.fab.import.course"),
```
[the remark this answers is not in the document]
Comment (psachmann) [reply src/components/templates/RoomWrapper.vue]: Is now resolved

File: src/serverApi/v3/api.ts [hunk 4]
```diff
@@ -10509,6 +10509,51 @@ export const CoursesApiAxiosParamCreator = function (configuration?: Configurati
...
+            };
+        },
+        /**
+         * 
+         * @summary Imports a course from a Common Cartridge file.
+         * @param {any} file The Common Cartridge file to import.
+         * @param {*} [options] Override http request option.
+         * @throws {RequiredError}
+         */
+        courseControllerImportCourse: async (file: any, options: any = {}): Promise<RequestArgs> => {
```
Comment (mkreuzkam-cap) [inline src/serverApi/v3/api.ts]: This method is now duplicated

**Review by mkreuzkam-cap — APPROVED** [review]

**Review by mkreuzkam-cap — APPROVED** [review]

**Pull request merged**